packages/core/src/utils/__tests__/indyProofRequest.test.ts

```diff
@@ -0,0 +1,240 @@
+import type { IndyCredentialMetadata } from '../../modules/credentials/models/CredentialInfo'
+import type { CustomCredentialTags, CredentialPreviewAttribute, RequestCredentialMessage } from '@aries-framework/core'
+
+import { checkProofRequestForDuplicates } from '..'
+import { Attachment, AttachmentData } from '../../decorators/attachment/Attachment'
+
+import {
+  CredentialMetadataKeys,
+  AriesFrameworkError,
+  AttributeFilter,
+  PredicateType,
+  ProofAttributeInfo,
+  ProofPredicateInfo,
+  ProofRequest,
+  CredentialState,
+  OfferCredentialMessage,
+  CredentialRecord,
+  CredentialPreview,
+} from '@aries-framework/core'
+
+export const INDY_CREDENTIAL_OFFER_ATTACHMENT_ID = 'libindy-cred-offer-0'
+
+const credentialPreview = CredentialPreview.fromRecord({
+  name: 'John',
+  age: '99',
+})
+
+const offerAttachment = new Attachment({
+  id: INDY_CREDENTIAL_OFFER_ATTACHMENT_ID,
+  mimeType: 'application/json',
+  data: new AttachmentData({
+    base64:
+      'eyJzY2hlbWFfaWQiOiJhYWEiLCJjcmVkX2RlZl9pZCI6IlRoN01wVGFSWlZSWW5QaWFiZHM4MVk6MzpDTDoxNzpUQUciLCJub25jZSI6Im5vbmNlIiwia2V5X2NvcnJlY3RuZXNzX3Byb29mIjp7fX0',
+  }),
+})
+
+const mockCredentialRecord = ({
+  state,
+  requestMessage,
+  metadata,
+  threadId,
+  connectionId,
+  tags,
+  id,
+  credentialAttributes,
+}: {
+  state?: CredentialState
+  requestMessage?: RequestCredentialMessage
+  metadata?: IndyCredentialMetadata & { indyRequest: Record<string, unknown> }
+  tags?: CustomCredentialTags
+  threadId?: string
+  connectionId?: string
+  id?: string
+  credentialAttributes?: CredentialPreviewAttribute[]
+} = {}) => {
+  const offerMessage = new OfferCredentialMessage({
+    comment: 'some comment',
+    credentialPreview: credentialPreview,
+    offerAttachments: [offerAttachment],
+  })
+
+  const credentialRecord = new CredentialRecord({
+    offerMessage,
+    id,
+    credentialAttributes: credentialAttributes || credentialPreview.attributes,
+    requestMessage,
+    state: state || CredentialState.OfferSent,
+    threadId: threadId ?? offerMessage.id,
+    connectionId: connectionId ?? '123',
+    tags,
+  })
+
+  if (metadata?.indyRequest) {
+    credentialRecord.metadata.set(CredentialMetadataKeys.IndyRequest, { ...metadata.indyRequest })
+  }
+
+  if (metadata?.schemaId) {
+    credentialRecord.metadata.add(CredentialMetadataKeys.IndyCredential, {
+      schemaId: metadata.schemaId,
+    })
+  }
+
+  if (metadata?.credentialDefinitionId) {
+    credentialRecord.metadata.add(CredentialMetadataKeys.IndyCredential, {
+      credentialDefinitionId: metadata.credentialDefinitionId,
+    })
+  }
+
+  return credentialRecord
+}
+
+describe('Present Proof', () => {
+  let secCredDef: CredentialRecord
+  let credDef: CredentialRecord
+
+  beforeAll(async () => {
+    credDef = mockCredentialRecord({ state: CredentialState.OfferSent })
+    secCredDef = mockCredentialRecord({ state: CredentialState.OfferSent })
```

Comment:
Because we're doing unit tests we can just create a random credential definition id (took this one from https://indyscan.io).

Also because the `checkProofRequestForDuplicates` method doesn't use the credential definition at all, we don't have to provide a second credential definition id. This should suffice (we also don't need the `mockCredentialRecord` etc... as the credential record is not used, we only need an id):

```suggestion
    const credentialDefinitionId = '9vPXgSpQJPkJEALbLXueBp:3:CL:57753:tag1'
```

Reply:
Do we still want to test if it passes with an attribute and a predicate with different credential id's? or should I remove those tests then as well?

Reply:
I think that could be useful, but this method doesn't test that. It never touches the credentialDefinitionId (note credentialDefinitionId is different from the credentialId. )


We should do an actual proof exchange to be able to test that (and should be done in proofs.test.ts)